### PR TITLE
Sync Intel ReduceOpToLLVM.cpp with the common version.

### LINKED ISSUE #795: [test_core.py::test_reduce_layouts] Fails to run on PVC
Wee need to enable tets currently skipped by this condition:
```
    if is_xpu() and type(src_layout) is BlockedLayout and src_layout.warps_per_cta == [2, 2]:
        pytest.skip("FIXME: incorrect result on XPU")
```

### PR DESCRIPTION
**Pull request by ienkovich**
Resolves #795

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp [hunk 1]
```diff
@@ -304,80 +310,50 @@ struct ReduceOpConversion
...
     auto mod = op.getOperation()->getParentOfType<ModuleOp>();
-    unsigned threadsPerWarp =
-        triton::gpu::TritonGPUDialect::getThreadsPerWarp(mod);
-    assert(threadsPerWarp > 1 &&
-           "threadsPerWarp must be larger than 1 to do warp shuffle reduce.");
     unsigned numThreads =
         product<unsigned>(triton::gpu::getWarpsPerCTA(srcLayout)) *
-        threadsPerWarp;
-
-    // It is a batched reduce with the initial problem shape [elems /
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: These changes are added intentionally in https://github.com/intel/intel-xpu-backend-for-triton/pull/697.

Comment (ienkovich) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: Which test does test these changes?

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: @ienkovich Do the test cases still pass if we sync with the common version plus adding the changes made in https://github.com/intel/intel-xpu-backend-for-triton/pull/697?

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: The change was added for the case when threads_per_warp < num_warps, e.g., changing threads_per_warp to 16.

Comment (ienkovich) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: The patch doesn't apply to the current common code and with no test, it's easy to make a mistake in porting. Also, these changes were added when `test_reduce_layouts` didn't use the proper compilation pipeline and therefore weren't properly tested. Wouldn't it be better to resubmit them with a test case added.

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: I am fine with that.

Comment (chengjunlu) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:323]: Yes. @ienkovich is right about it. Let's go ahead. And I will submit a new PR with the unit test cases for the issues here in the common code.

**Review by chengjunlu — APPROVED** [review]

**Pull request merged**